source/postprocess/stokes_residual.cc

```diff
@@ -0,0 +1,164 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/stokes_residual.h>
+#include <aspect/simulator_access.h>
+#include <aspect/simulator.h>
+#include <aspect/global.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/fe/fe_dgq.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/numerics/data_out_stack.h>
+
+
+#include <math.h>
+#include <vector>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    template <int dim>
+    template <class Archive>
+    void StokesResidual<dim>::DataPoint::serialize (Archive &ar,
+                                                    const unsigned int)
+    {
+      ar &time &solve_index &values;
+    }
+
+
+    template <int dim>
+    StokesResidual<dim>::StokesResidual ()
+    {}
+
+
+
+
+    template <int dim>
+    std::pair<std::string,std::string>
+    StokesResidual<dim>::execute (TableHandler &)
+    {
+      // On the root process, write out the file.
+      if (Utilities::MPI::this_mpi_process(this->get_mpi_communicator()) == 0)
+        {
+          std::ofstream f((this->get_output_directory() +
+                           "stokes_residuals.txt").c_str());
+          f << "# time solveidx residual\n";
+          for (unsigned int i=0; i<entries.size(); ++i)
+            {
+              for (unsigned int j=0; j<entries[i].values.size(); ++j)
+                f << entries[i].time << " "
+                  << entries[i].solve_index << " "
```

Comment:
You can get a single space with `' '`. Not that it's going to make a whole lot of difference, but it will save you some cycles!